models/connection/message.go

```diff
@@ -4,7 +4,7 @@ type NoPayload bool
 type Message[T any] struct {
 	Code    int     `json:"code"`
 	Payload T       `json:"payload,omitempty"`
-	Error   RespErr `json:"error,omitempty"`
+	Error   *RespErr `json:"error,omitempty"`
```

Comment:
this is not related to this PR, in the future let's keep the changes related to the branch/PR. Having related branch/PR helps with testing, clarity and reverting if it was ever needed